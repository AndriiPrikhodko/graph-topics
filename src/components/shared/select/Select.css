.select {
    width: '100px';
    height: '30px';
    padding: '5px';
    font-size: '14px';
    background-color: lightgrey;
    border: '1px solid #ccc';
    border-radius: '4px';
}