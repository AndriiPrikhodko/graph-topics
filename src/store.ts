import { configureStore } from '@reduxjs/toolkit'
import rootReducer from './reducers'
import thunk from 'redux-thunk'

const store = configureStore({
  reducer: {
    graphData: rootReducer.graphSlice,
    graphList: rootReducer.graphListSlice
  },
  middleware: [thunk],
})

export default store;