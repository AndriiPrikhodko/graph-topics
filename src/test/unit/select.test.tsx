import { vi, expect } from 'vitest'
import { test } from 'vitest'
import { render } from '@testing-library/react'
import store from '../../store'
import { Provider } from 'react-redux'
import Select from '../../components/shared/select'
import { fireEvent } from '@testing-library/react'

test('menu select link strategy chain', async () => {

    const Props = {
        label: '',
        'testid': 'link-strategy',
        onChange: vi.fn(),
        optionList: [
            'from',
            'chain'
        ]
    }

    const selectR = render(
        <Provider store={store}>
            <Select {...Props}/>
        </Provider>
    )

    const select = selectR.getByTestId('link-strategy')

    expect(select).not.toBeNull()

    if(select)
        await fireEvent.change(select, { target: { value: 'chain' } })

    expect(Props.onChange).toHaveBeenCalled()
    expect(select).toMatchSnapshot()
})