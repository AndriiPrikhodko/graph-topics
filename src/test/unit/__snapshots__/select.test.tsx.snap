// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`menu select link strategy chain 1`] = `
<select
  class="select "
  data-testid="link-strategy"
  title=""
>
  <option
    data-testid="link-from"
    value="from"
  >
    from
  </option>
  <option
    data-testid="link-chain"
    value="chain"
  >
    chain
  </option>
</select>
`;
